service/datasource-service.py

```diff
@@ -167,38 +169,109 @@ def decorated(*args, **kwargs):
 
     return decorated
 
-def generate_sheetdata(url,auth,params,headers,sheets,ids,names,direction,start,since):
-    logger.info("downloading from url=%s with params=%s" % (url, params))
-    r = requests.get(url, auth=auth, headers=headers, params=params)
-    r.raise_for_status()
-    logger.debug("Got file data")
-    workbook = xlrd.open_workbook("sesm.xsls", file_contents=r.content)
 
+def generate_sheetdata(workbook, vars):
+    sheet = vars.get('sheet', get_envvar('sheet')) or '1'
+    sheets = [int(x) - 1 for x in sheet.split(',')]
+    ids = vars.get('ids', get_envvar('ids')) or "0"
+    ids = [int(x) - 1 for x in ids.split(',')]
+    names = vars.get('names', get_envvar('names')) or "1"
+    names = [int(x) - 1 for x in names.split(',')]
+    direction = vars.get('direction', get_envvar('direction')) or "row"
+    start = vars.get('start', get_envvar('start'))
+    since = vars.get('since', "0001-01-01")
+
+    if not start and direction == "row":
+        start = "1,2"
+    elif not start:
+        start = "2,1"
+
+    start = [int(x) - 1 for x in start.split(',')]
     modified = "modified" in workbook.props
+
     if modified:
         modified = workbook.props["modified"]
 
     if not modified or modified > since:
         iterable_sheets = sheets or range(0, workbook.nsheets)
         for sheet_index in iterable_sheets:
             worksheet = workbook.sheet_by_index(sheet_index)
-            id_prefix = str(sheet_index+1) + "-" if len(iterable_sheets)>1 else ""
+            id_prefix = str(sheet_index+1) + "-" if len(iterable_sheets) > 1 else ""
             if direction == "row":
-                columnNames = getColNames(worksheet, names, start)
-                yield from getSheetRowData(worksheet, columnNames, start, ids, modified, workbook.datemode, id_prefix)
+                column_names = getColNames(worksheet, names, start)
+                yield from getSheetRowData(worksheet, column_names, start, ids, modified, workbook.datemode, id_prefix)
             else:
-                rowNames = getRowNames(worksheet, names, start)
-                yield from getSheetColData(worksheet, rowNames, start, ids, modified, workbook.datemode, id_prefix)
+                row_names = getRowNames(worksheet, names, start)
+                yield from getSheetColData(worksheet, row_names, start, ids, modified, workbook.datemode, id_prefix)
+
+
+def extract_content(entity):
+    """
+    Given an entity, attempt to find "content" (encoded bytes) and "content-type" (Mime type of file).
+    """
+
+    content = None
+    content_type = None
+
+    for key in entity.keys():
+        if "content" in key.split(":"):
```

Comment:
This will also try to use the property "content:foo" if the namespace happens to be "content". Not sure if the microservice should try to strip off namespaces, but it's not a very big deal.